Sync and add missing 'readonly' in [Audio|Video|Text]Track.idl as per web specification

Ahmad-S792 opened the pull request:
#### 93d70016b8882ffa3490fa3117ea60eb6f5ce8b3
<pre>
Sync and add missing &apos;readonly&apos; in [Audio|Video|Text]Track.idl as per web specification

<a href="https://bugs.webkit.org/show_bug.cgi?id=262060">https://bugs.webkit.org/show_bug.cgi?id=262060</a>

Reviewed by Chris Dumez and Eric Carlson.

This PR adds missing `readonly` for &apos;kind&apos; and &apos;language&apos; as
required by web specification [1], [2], [3], it was missed
unintentionally.

[1] <a href="https://html.spec.whatwg.org/#texttrack">https://html.spec.whatwg.org/#texttrack</a>
[2] <a href="https://html.spec.whatwg.org/#audiotrack">https://html.spec.whatwg.org/#audiotrack</a>
[3] <a href="https://html.spec.whatwg.org/#videotrack">https://html.spec.whatwg.org/#videotrack</a>

* Source/WebCore/html/track/TextTrack.idl:
* Source/WebCore/html/track/VideoTrack.idl:
* Source/WebCore/html/track/AudioTrack.idl:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Rebaselined
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Rebaselined
* LayoutTests/media/media-source/only-bcp47-language-tags-accepted-as-valid.html: Rebaselined
* LayoutTests/media/media-source/only-bcp47-language-tags-accepted-as-valid-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/268580@main">https://commits.webkit.org/268580@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/08fac4350c61cf4323a2f20c3b9ccc1524103bbc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20071 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20510 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21128 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21970 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18736 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20301 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23753 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20668 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20219 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20291 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20214 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17448 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22820 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17384 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18245 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24497 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18461 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18421 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22487 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19012 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16140 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18208 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18081 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4817 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22551 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18837 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/html/track/AudioTrack.idl:
```diff
@@ -32,9 +32,9 @@
 ] interface AudioTrack {
     readonly attribute DOMString id;
-    [EnabledConditionallyReadWriteBySetting=MediaSourceEnabled] attribute [AtomString] DOMString kind;
+    [EnabledConditionallyReadWriteBySetting=MediaSourceEnabled] readonly attribute [AtomString] DOMString kind;
```
EnabledConditionallyReadWriteBySetting seems to have been added specifically for these attributes. It does seem like it was added by mistake as there's no need for these to be publicly writable, but then we should completely remove it here.

Ahmad-S792 replied:
Done and removed.

cdumez reviewed (approved):
LGTM

Ahmad-S792 committed:
Sync and add missing 'readonly' in [Audio|Video|Text]Track.idl as per web specification

https://bugs.webkit.org/show_bug.cgi?id=262060

Reviewed by Chris Dumez and Eric Carlson.

This PR adds missing `readonly` for 'kind' and 'language' as
required by web specification [1], [2], [3], it was missed
unintentionally.

[1] https://html.spec.whatwg.org/#texttrack
[2] https://html.spec.whatwg.org/#audiotrack
[3] https://html.spec.whatwg.org/#videotrack

* Source/WebCore/html/track/TextTrack.idl:
* Source/WebCore/html/track/VideoTrack.idl:
* Source/WebCore/html/track/AudioTrack.idl:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Rebaselined
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Rebaselined
* LayoutTests/media/media-source/only-bcp47-language-tags-accepted-as-valid.html: Rebaselined
* LayoutTests/media/media-source/only-bcp47-language-tags-accepted-as-valid-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/268580@main
